app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -668,54 +611,31 @@ class UploadSupportingEvidenceController @Inject() (
 
   def deleteSupportingEvidence(id: String): Action[AnyContent] = authenticatedActionWithSessionData.async {
     implicit request =>
-      withUploadSupportingEvidenceAnswers(request) { (draftReturnId, _, _, fillingOutReturn, answers) =>
-        val result = for {
-          _ <- upscanService.removeFile(DraftReturnId(draftReturnId.toString), UpscanInitiateReference(id))
-          updatedAnswers = answers match {
-            case IncompleteUploadSupportingEvidenceAnswers(
-                doYouWantToUploadSupportingEvidenceAnswer,
-                supportingEvidences
-                ) =>
+      withUploadSupportingEvidenceAnswers(request) { (_, _, _, fillingOutReturn, answers) =>
+        val updatedAnswers = answers.fold(
+          incomplete =>
+            incomplete.copy(
+              evidences        = incomplete.evidences.filterNot(_.reference === id),
+              expiredEvidences = incomplete.expiredEvidences.filterNot(_.reference === id)
+            ), { complete =>
+            val newEvidences = complete.evidences.filterNot(_.reference === id)
+            if (newEvidences.isEmpty)
```

Comment:
if we're complete with one document and we delete the document, I made it so that we go back to incomplete since it means we need to upload a document again since we can't have no documents after saying yes